producer-rest-core/src/main/scala/za/co/absa/spline/producer/rest/controller/ErrorControllerAdvice.scala

```diff
@@ -16,14 +16,22 @@
 
 package za.co.absa.spline.producer.rest.controller
 
-import org.springframework.http.HttpStatus.INTERNAL_SERVER_ERROR
+import org.springframework.beans.TypeMismatchException
+import org.springframework.http.HttpStatus.{BAD_REQUEST, INTERNAL_SERVER_ERROR}
 import org.springframework.http.ResponseEntity
+import org.springframework.http.converter.HttpMessageConversionException
 import org.springframework.web.bind.annotation.{ControllerAdvice, ExceptionHandler}
 import za.co.absa.spline.common.logging.ErrorCode
 
 @ControllerAdvice(basePackageClasses = Array(classOf[_package]))
 class ErrorControllerAdvice {
 
+  @ExceptionHandler(Array(
+    classOf[TypeMismatchException],
+    classOf[HttpMessageConversionException]
+  ))
+  def handle_400(e: Exception) = new ResponseEntity(ErrorCode(e), BAD_REQUEST)
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Method name does not match the regular expression '^([a-z][A-Za-z0-9]*(?:_=)?|<<|>>>|>|==|!=|<|<=|>|>=|\||&|\^|\+|-|\*|\/|%|\|\||&&|\+\+|--|\+=|-=)$'](https://app.codacy.com/app/ABSA_OSS/spline/pullRequest?prid=4536902)